db/data_migration/20180806082233_redirect_gambia_news.rb

```diff
@@ -0,0 +1,6 @@
+puts "Unpublishing /world/gambia/news and redirecting to /world/the-gambia/news"
+Services.publishing_api.unpublish(
+  "3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754",
+  type: "redirect",
+  alternative_path: "/world/the-gambia/news",
+)
```

Comment:
Unpublishing is only if the content hasn't moved, but if a new bit has been created. That would be a bit odd in the case of renaming a country, but is that the case?

Reply:
The country was reslugged here: https://github.com/alphagov/whitehall/blob/master/db/data_migration/20180731114637_rename_gambia.rb

Which seems to have resulted in it being published as a new thing.

Reply:
Ok... do you know what things in Whitehall (if any) correspond to the pages in question here? The content_id of the WorldLocation that's updated in that data migration doesn't match the content_id's for the .../news pages in the Publishing API. I'm wondering if there are some more things in Whitehall that need changing...

```
publishing_api_production=# SELECT documents.id, content_id, editions.id, editions.base_path FROM editions INNER JOIN documents ON documents.id = editions.document_id WHERE base_path = '/world/the-gambia/news' OR base_path = '/world/gambia/news';
   id   |              content_id              |   id    |       base_path        
--------+--------------------------------------+---------+------------------------
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2613830 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2743299 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2743760 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2636011 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2842060 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2842383 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2797393 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 2797396 | /world/gambia/news
 693058 | 3f8c93e2-5da5-4ba8-80ad-e10b0cc5a754 | 3168854 | /world/gambia/news
 822503 | 34e15756-b2af-4d13-a2ba-d1b1c8c4ae5e | 3280134 | /world/the-gambia/news
 822503 | 34e15756-b2af-4d13-a2ba-d1b1c8c4ae5e | 3280137 | /world/the-gambia/news
(11 rows)
```

Reply:
I think that'll be the slug for the old page.  `34e15756-b2af-4d13-a2ba-d1b1c8c4ae5e` matches the slug in whitehall for the new WorldLocation:

```
irb(main):005:0> WorldLocation.find_by(slug: "the-gambia").content_id
=> "5e9f010f-7706-11e4-a3cb-005056011aef"
```

It looks like the worker for publishing world location news pages fetches the content ID from the content store by base path, rather than using the content ID in the whitehall database: https://github.com/alphagov/whitehall/blob/720c7dfc65c336349ac50da0624b3b3094498b61/app/workers/world_location_news_page_worker.rb#L42-L48
In which case, changing the slug would cause it to generate a new content ID, because the content store won't return anything.

Reply:
Huh, that is very odd. They've had content_id's for a long time ( 083304c637c5ff3550ab4580fb532b90c79e7832 ), but I guess there's been some drift because of this lookup and pick randomly code... 

So yeah, it's looking like unpublishing this old content is a workaround. It would be good to fix this worker to prevent this from happening again though.